simple-facebook-instant-articles.php

```diff
@@ -405,10 +406,65 @@ public function api_galleries_shortcode( $atts ) {
 	 * @return string           FB IA formatted markup for social embeds.
 	 */
 	public function reformat_social_embed( $html, $url, $attr, $post_ID = null ) {
+
 		return sprintf( '<figure class="op-social"><iframe>%s</iframe></figure>', $html );
 	}
 
 	/**
+	 * Ensure facebook scripts are loaded for facebook embeds.
+	 *
+	 * @param string   $html    HTML markup to be embeded into post content.
+	 * @param string   $url     The attempted embed URL.
+	 * @param array    $attr    An array of shortcode attributes.
+	 *
+	 * @return string           Facebook embed code with required script.
+	 */
+	public function load_facebook_scripts( $html, $url, $attr ) {
+		global $wp_embed;
+
+		// Use handlers IDs as keys for quicker lookup.
+		$facebook_handlers_ids = array(
+			'facebook'                 => true,
+			'facebook-alternate'       => true,
+			'facebook-photo'           => true,
+			'facebook-alternate-photo' => true,
+			'facebook-video'           => true,
+		);
+
+		$is_facebook_embed = false;
+
+		// Check if we're processing facebook embed.
+		foreach ( $wp_embed->handlers as $priority => $handlers ) {
+			foreach ( $handlers as $name => $args ) {
+
+				// Stop - if we looked at all FB handlers.
+				if ( ! $facebook_handlers_ids ) {
+					break 2;
+				}
+				// See if embed URL is one of facebook recognised embed URLs.
+				elseif ( isset( $facebook_handlers_ids[ $name ] ) ) {
+
+					if ( preg_match( $args['regex'], $url ) ) {
+						$is_facebook_embed = true;
+						break 2;
+					}
+
+					unset( $facebook_handlers_ids[ $name ] );
+				}
+			}
+		}
+
+		// It is a facebook embed - insert script.
+		if ( $is_facebook_embed && function_exists( 'jetpack_facebook_add_script' ) ) {
```

Comment:
might be best to have the function exists check right at the top of the function? No point in doing the whole find the embed handler dance if the function doesn't exist.